### DISCUSSION TITLE
How you create new models for tabular input?

### DISCUSSION THREAD

**Discussion by WinterSilence** [yiisoft/yii2#19279] [Q&A]
I never using `*Multiple` methods before. My back-end forms communicates with Yii application through AJAX requests to REST API. API methods work with single models(composite or not).
After reading [Creating a dynamic set of new records](https://www.yiiframework.com/doc/guide/2.0/en/input-tabular-input#creating-a-dynamic-set-of-new-records) I thought: what's strange solution... why need create empty models? why not just clone prototype model?!
~~~php
$setting = Setting::instance();
$setting->ensureBehaviors(); // generate behaviors
$setting-createValidators(); // and validators
$settings = [];
while (--$count > 0) {
   $settings[] = clone $setting;
}
~~~
I look at how cloning is implemented in models and was unpleasantly surprised: `Model` remove event handlers and `Behavior`s instead cloning, but other properties containing objects are ignored. Handlers/behaviours can be re-attached to cloned model without any problems.
Aslo, `Model::instance()` not calls `ensureBehaviors()`, `createValidators()` and return not completely initiated model.

**How do you solve such problems?**

**Comment by samdark** [comment]
Why do you prefer cloning in this case? Performance or something else?

**Reply by WinterSilence** [reply]
Performance - every model contains many objects(validators and behaviors)

**Reply by samdark** [reply]
OK. Makes sense.

**Reply by samdark** [reply]
I never had the need for it performance-wise, that's why I never solved such issue.

**Reply by WinterSilence** [reply]
@samdark sounds like as unreal - any complex application have forms like this:
![изображение](https://user-images.githubusercontent.com/3521094/162412320-54f6e075-c9f8-4496-a459-f6938648e651.png)

**Reply by samdark** [reply]
Not my type of applications, it seems...

**Reply by WinterSilence** [reply]
@samdark you don't have applications with relations "one to many" like as item images or user roles? you kidding me :(
